web-api/src/mind.rs

```diff
@@ -402,49 +404,52 @@ async fn run_persona_benchmark() -> Result<(), Error> {
 }
 
 /// Runs the user-based mind benchmark
-#[test]
+#[tokio::test]
 #[ignore]
-fn run_user_benchmark() -> Result<(), Error> {
+async fn run_user_benchmark() -> Result<(), Error> {
     let document_provider = DocumentProvider::new("news.tsv")?;
 
-    let impressions = read("behaviors.tsv")?;
+    let state = State::new(Storage::default()).unwrap();
+    state
+        .insert(document_provider.to_documents())
+        .await
+        .unwrap();
 
     let nranks = vec![3];
     let mut ndcgs = Array::zeros((nranks.len(), 0));
 
     // Loop over all impressions, prepare reranker with news in click history
     // and rerank the news in an impression
-    for impression in impressions {
+    for impression in read("behaviors.tsv")? {
         let impression: Impression = impression?;
+        let user = UserId::new(impression.user_id).unwrap();
 
-        // Placeholder for interacting with the entire click history
-        for click in impression.clicks {
-            match document_provider.get(&click) {
-                Some(document) => println!("The document {:?} was interacted.", document),
-                None => println!("Document id {} not found.", click),
-            }
-        }
+        state.interact(&user, &impression.clicks).await.unwrap();
 
-        // Placeholder for reranking the results
-        let mut snippet_label_pairs = impression
+        let document_ids = impression
             .news
             .iter()
-            .filter_map(|viewed_document| {
-                document_provider
-                    .get(&viewed_document.document_id)
-                    .map(|document| {
-                        SnippetLabelPair(document.snippet.clone(), viewed_document.was_clicked)
-                    })
-            })
-            .collect_vec();
-        snippet_label_pairs.shuffle(&mut thread_rng());
+            .map(|document| &document.document_id)
+            .collect::<Vec<_>>();
 
-        let labels = snippet_label_pairs
+        let labels = state
+            .personalize(&user, PersonalizeBy::Documents(document_ids.as_slice()))
+            .await
+            .unwrap()
             .iter()
-            .map(|snippet_label| if snippet_label.1 { 1.0 } else { 0.0 })
+            .map(|reranked_id| {
+                u8::from(
+                    impression.news[document_ids
+                        .iter()
+                        .position(|&actual_id| actual_id == reranked_id)
+                        .unwrap()]
+                    .was_clicked,
+                )
+                .into()
```

Comment:
I have been using `2.0` as a value for "clicked" news, if I understand correctly this code will return `1.0` for clicked articles. For the representative results, we should agree on a single value in both tests. I advocate for 2 because we have been using it in previous tests performed in python and we could maybe compare some results. WDYT?

Reply:
Although always tending towards consistency, I am against it in this case as the results of the user benchmark are not comparable to other externals anymore. Doesn't matter that much for the persona-related one, do we have any representative results in python documented somewhere that we could use reference?